### ISSUE TITLE
Odd behaviour when training with maxEpochs=1

### ISSUE THREAD

**Issue by dmagee** [Microsoft/CNTK#406]
I'm playing with a slightly modified version of the CFAR-10 example and setting maxEpochs=1 [**see below for why I want to do this]. Under certain circumstances I'm getting an error:

Starting minibatch loop.
 Epoch[ 1 of 1]-Minibatch[   1- 500]: SamplesSeen = 500; TrainLossPerSample =  5.87155566; EvalErr[0]PerSample = 0.27600000; TotalTime = 2.7641s; SamplesPerSecond = 180.9
 Epoch[ 1 of 1]-Minibatch[ 501-1000]: SamplesSeen = 500; TrainLossPerSample =  1.#QNAN000; EvalErr[0]PerSample = 0.48000000; TotalTime = 2.3504s; SamplesPerSecond = 212.7

[CALL STACK]
    >Microsoft::MSR::CNTK::TracingGPUMemoryAllocator::operator=
    -Microsoft::MSR::CNTK::TracingGPUMemoryAllocator::operator=
    -Microsoft::MSR::CNTK::TracingGPUMemoryAllocator::operator=
    -Microsoft::MSR::CNTK::TracingGPUMemoryAllocator::operator=
    -Microsoft::MSR::CNTK::TracingGPUMemoryAllocator::operator=
    -Microsoft::MSR::CNTK::TracingGPUMemoryAllocator::operator=
    -Microsoft::MSR::CNTK::TracingGPUMemoryAllocator::operator=
    -Microsoft::MSR::CNTK::TracingGPUMemoryAllocator::operator=
    -Microsoft::MSR::CNTK::TracingGPUMemoryAllocator::operator=
    -Microsoft::MSR::CNTK::TracingGPUMemoryAllocator::operator=
    -BaseThreadInitThunk
    -RtlUserThreadStart

EXCEPTION occurred: The training criterion is not a number (NAN). Stop

With maxEpochs=2 (or more) everything is fine, and logically surely doing the first of 2 epochs is the same as doing 1 epoch? Anyway, on further investigation changing the parameters in the example makes a difference:

```
    learningRatesPerMB=0.01*10:0.003*10:0.001 
    momentumPerMB=0.9*20:0.99 
    dropoutRate=0*5:0.5 
```

The theory was that it didn't make sense to use _5,_10,*20 when there was only one epoch (although it seems to work fine with 2 epochs), so changing one of the parameters to:

learningRatesPerMB=0.01

All works fine... BUT logically if you need to change that parameter, you should need to change the other 2 as well right?

```
    learningRatesPerMB=0.01 
    momentumPerMB=0.9 
    dropoutRate=0 
```

That's where it gets odd, as that reintroduces the problem! This is all pretty deterministic (I tried each several times). I know it's not a huge bug, but I do think something odd is going on.

Derek

*\* So why do I want to do something stupid like this? I have a task scheduler so different jobs can share the machine fairly (i.e. not wait a day or two for a small job to run). Thus I want to break up training of larger jobs. Turns out cntk makes that quite easy by:
For e=1:NO_EPOCHS_REQUIRED:
    delete any previous model output (but not intermediate files)
    (over)write .cntk file with maxEpochs=e
    run cntk.exe
    check for error by checking final model file has been created

There may be a better way of doing this, but this works for me!

**Comment by dmagee** [comment]
Also, same behaviour with =2 on a different (larger) data set:

        learningRatesPerMB=0.001 
        momentumPerMB=0.9*20:0.99 
        maxEpochs=2
        dropoutRate=0*5:0.5 

...

Starting minibatch loop.
 Epoch[ 1 of 2]-Minibatch[   1- 500]: SamplesSeen = 500; TrainLossPerSample =  0.88958105; EvalErr[0]PerSample = 0.29800000; TotalTime = 62.4512s; SamplesPerSecond = 8.0
 Epoch[ 1 of 2]-Minibatch[ 501-1000]: SamplesSeen = 500; TrainLossPerSample =  0.85155786; EvalErr[0]PerSample = 0.30800000; TotalTime = 28.8722s; SamplesPerSecond = 17.3
 Epoch[ 1 of 2]-Minibatch[1001-1500]: SamplesSeen = 500; TrainLossPerSample =  1.#QNAN000; EvalErr[0]PerSample = 0.48600000; TotalTime = 33.1901s; SamplesPerSecond = 15.1

[CALL STACK]
    >Microsoft::MSR::CNTK::TracingGPUMemoryAllocator::operator=
    -Microsoft::MSR::CNTK::TracingGPUMemoryAllocator::operator=
    -Microsoft::MSR::CNTK::TracingGPUMemoryAllocator::operator=
    -Microsoft::MSR::CNTK::TracingGPUMemoryAllocator::operator=
    -Microsoft::MSR::CNTK::TracingGPUMemoryAllocator::operator=
    -Microsoft::MSR::CNTK::TracingGPUMemoryAllocator::operator=
    -Microsoft::MSR::CNTK::TracingGPUMemoryAllocator::operator=
    -Microsoft::MSR::CNTK::TracingGPUMemoryAllocator::operator=
    -Microsoft::MSR::CNTK::TracingGPUMemoryAllocator::operator=
    -Microsoft::MSR::CNTK::TracingGPUMemoryAllocator::operator=
    -BaseThreadInitThunk
    -RtlUserThreadStart

EXCEPTION occurred: The training criterion is not a number (NAN). Stop

**Comment by dmagee** [comment]
GFORCE GT 740M in a laptop. Certainly reducing the learning rate seemed to fix it, but randomly changing settings also fixes things / breaks things, so I'm not going to say this is a sure fire fix.

**Comment by wolfma61** [comment]
The issue has been dormant for a long while. Please open a new issue if you still experience problems in the latest release.
